03_cross_cancer_classification/plot_flip_labels_results.ipynb

```diff
@@ -0,0 +1,546 @@
+       "      <td>1</td>\n",
+       "      <td>0.2</td>\n",
+       "      <td>0.031522</td>\n",
+       "      <td>BRAF</td>\n",
+       "      <td>COAD</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>1</td>\n",
+       "      <td>0.3</td>\n",
+       "      <td>0.192914</td>\n",
+       "      <td>BRAF</td>\n",
+       "      <td>COAD</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>1</td>\n",
+       "      <td>0.5</td>\n",
+       "      <td>-0.019665</td>\n",
+       "      <td>BRAF</td>\n",
+       "      <td>COAD</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>1</td>\n",
+       "      <td>0.1</td>\n",
+       "      <td>0.006582</td>\n",
+       "      <td>BRAF</td>\n",
+       "      <td>LUAD</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   index  percent_flip      aupr  gene cancer_type\n",
+       "0      1           0.1  0.591023  BRAF        COAD\n",
+       "1      1           0.2  0.031522  BRAF        COAD\n",
+       "2      1           0.3  0.192914  BRAF        COAD\n",
+       "3      1           0.5 -0.019665  BRAF        COAD\n",
+       "4      1           0.1  0.006582  BRAF        LUAD"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "flip_norm_df = au.normalize_to_control(flip_labels_df,\n",
+    "                                       train_id='train_identifier',\n",
+    "                                       test_id='test_identifier',\n",
+    "                                       additional_cols=['percent_flip'])\n",
+    "flip_norm_df = (\n",
+    "    flip_norm_df.assign(gene=lambda x: x.train_identifier.str.split('_', expand=True)[0],\n",
+    "                        cancer_type=lambda x: x.train_identifier.str.split('_', expand=True)[1])\n",
+    "                .drop(columns=['train_identifier', 'test_identifier'])\n",
+    "                .reset_index()\n",
+    ")\n",
+    "flip_norm_df.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sns.set({'figure.figsize': (15, 10)})\n",
+    "fig, axarr = plt.subplots(2, 2)\n",
+    "\n",
+    "percents_flip = flip_norm_df.percent_flip.unique()\n",
+    "for ix, percent_flip in enumerate(percents_flip):\n",
+    "    ax = axarr[ix // 2, ix % 2] \n",
+    "    heatmap_df = (\n",
+    "        flip_norm_df.query('percent_flip == @percent_flip')\n",
+    "                    .pivot(index='gene', columns='cancer_type', values='aupr')\n",
+    "    )\n",
+    "    sns.heatmap(heatmap_df, vmin=0.0, vmax=1.0, ax=ax)\n",
+    "    ax.set_title('Percent of labels flipped: {}'.format(percent_flip))\n",
+    "plt.tight_layout()"
```

Comment:
I think there's supposed to be a heatmap here, but it doesn't show up when rendering the notebook

Reply:
Nice catch! I'll fix this.